url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/communication/xboard/EngineController.java

```diff
@@ -0,0 +1,513 @@
+/*
+ * Copyright (c) 2008-2011, David Garcinuño Enríquez <dagaren@gmail.com>
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package es.dagaren.gladiator.communication.xboard;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Map.Entry;
+
+import es.dagaren.gladiator.communication.ProtocolController;
+import es.dagaren.gladiator.notation.Notation;
+import es.dagaren.gladiator.representation.Movement;
+
+/**
+ * @author dagaren
+ * 
+ */
+public class EngineController extends ProtocolController implements
+      EngineToUser
+{
+   protected UserToEngine receiver;
+
+   /**
+    * @param receiver
+    *           the receiver to set
+    */
+   public void setReceiver(UserToEngine receiver)
+   {
+      this.receiver = receiver;
+   }
+
+   /**
+    * @return the receiver
+    */
+   public UserToEngine getReceiver()
+   {
+      return receiver;
+   }
+
+   @Override
+   public void onCommandReceived(String command)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 44. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.